src/components/CustomPrompt.js

```diff
@@ -0,0 +1,161 @@
+    textTransform: 'capitalize'
+  },
+  carousel: {
+    margin: '2em 0em',
+    '& .slide': {
+      display: 'grid',
+      placeItems: 'center'
+    },
+    '& .slider': {
+      maxHeight: '25em'
+    },
+    '& image': {
+      maxHeight: '25em'
+    }
+  }
+}))
+
+const CustomPrompt = ({
+  appName,
+  appLinks = [],
+  appImages = [],
+  appIcon,
+  description,
+  learnMoreUrl
+}) => {
+  const needsLearnMore = description.includes('\n')
+
+  const classes = useStyles()
+  const [showFullDescription, setShowFullDescription] = useState(false)
+
+  const transformedDescription = showFullDescription ? description : description.split('\n')[0]
+
+  return (
+    <div>
+      <div className={classes.top_area}>
+        <img src={appIcon} className={classes.app_icon} />
+        <div className={classes.right_container}>
+          <div>
+            <Typography style={{ fontWeight: 500 }} variant='h4'>
+              {appName}
+            </Typography>
+            {appLinks.map((link, i) => link.length > 1
+              ? (
+                <Link
+                  key={i}
+                  href={link[1]}
+                  style={{ cursor: 'pointer', marginRight: '1em' }}
+                  target='_blank'
+                >
+                  {link[0]}
+                </Link>
+                )
+              : (
+                <Typography color='textSecondary' className={classes.secondary}>
+                  {link[0]}
+                </Typography>
+                ))}
+          </div>
+          <div className={classes.bottom_right_container}>
+            <div>
+              <Typography
+                style={{
+                  // fontSize: '0.8em'
+                  fontWeight: '300'
+                }}
+              >
+                <Info
+                  color='primary'
+                  style={{
+                    opacity: 0.6,
+                    height: '0.75em'
+                  }}
+                />
+                This app requires Babbage Desktop
+              </Typography>
+            </div>
+            <Button
+              color='primary'
+              className={classes.install_button}
+              variant='contained'
+              size='small'
+              onClick={() => {
+                window.open('https://projectbabbage.com/desktop#launch', '_blank')
+              }}
+            >
+              Install Babbage
+            </Button>
+          </div>
+        </div>
+      </div>
+      <Carousel
+        showIndicators={false}
+        showThumbs={false}
+        showStatus={false}
+        className={classes.carousel}
+        emulateTouch
+      >
+        {appImages.map((url, i) => (
+          <div key={i}>
+            <img src={url} />
+          </div>
+        ))}
+      </Carousel>
+      <Typography style={{ marginBottom: '2em', whiteSpace: 'pre-line' }} paragraph>
+        {transformedDescription}
+        <br />
+        {learnMoreUrl && needsLearnMore && (
```

Comment:
```suggestion
        {needsLearnMore && (
```
I think the Learn More should be a toggle and should not have a URL